horovod/spark/lightning/remote.py

```diff
@@ -223,8 +228,9 @@ def on_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -
                     # One more file sync to push profiler result.
                     remote_store.sync(logs_path)
 
-                # rank 0 overwrites model with best checkpoint and returns.
-                best_model = model.load_from_checkpoint(_checkpoint_callback.best_model_path)
+                # Rank 0 overwrites model with best checkpoint and returns.
+                best_model = model if use_last_model_weights else model.load_from_checkpoint(_checkpoint_callback.best_model_path, strict=False)
```

Comment:
best_model = model if not _checkpoint_callback else ...

Reply:
_checkpoint_callback is always not None because of the line 141